src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,90 @@
-import { useState } from 'react';
+import React, { FormEvent, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isValidData = (
+    !title.trim()
+    || !imgUrl.trim()
+    || !imdbUrl.trim()
+    || !imdbId.trim()
+  );
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    setCount(count + 1);
```

Comment:
here should use callback form too )
https://mate.academy/learn/react/updating-state#/video/318
```suggestion
    setCount(count => count + 1);
```